ingest/anndata_.py

```diff
@@ -127,3 +133,92 @@ def delocalize_extracted_files(file_path, study_file_id, files_to_delocalize):
                 file,
                 f"_scp_internal/anndata_ingest/{study_file_id}/{file}",
             )
+
+    def process_matrix(self):
+        """Perform matrix processing"""
+        if self.feature_names_unique(self.adata.var_names):
+            self.transform()
+
+    @staticmethod
+    def feature_names_unique(features):
+        """Return True if feature names are unique, else false"""
+        if len(features) == len(features.unique()):
+            return True
+        else:
+            # placeholder for function to provide duplicate feature names
+            # non-happy path - option to use gene IDs?
+            msg = f"Duplicate feature name found"
+            GeneExpression.log_for_mixpanel(
+                "error", "content:duplicate:values-within-file", msg
+            )
+            raise ValueError(msg)
+
+    def transform(self):
+        """Transforms matrix into gene data model."""
+        # initialize settings for mock data loads in tests
+        self.test_models = None
+        self.models_processed = 0
+
+        # derive file name from file path
+        file_name = os.path.basename(self.file_path)
+        start_time = datetime.datetime.now()
+        GeneExpression.dev_logger.info("Starting run at " + str(start_time))
+        num_processed = 0
+        gene_models = []
+        data_arrays = []
+        for all_cell_model in GeneExpression.create_data_arrays(
+            name=f"{file_name} Cells",
+            array_type="cells",
+            values=self.adata.obs.index.tolist(),
+            linear_data_type="Study",
+            linear_data_id=self.study_file_id,
+            **self.data_array_kwargs,
+        ):
+            data_arrays.append(all_cell_model)
+
+        # ASSUMPTION all_cell_model same for raw_count and processed_expression
+        # TODO: if raw counts is indicated check that .raw slot is populated
+
+        # Iterate over feature names (for happy path)
+        for feature in self.adata.var_names.tolist():
+            print(f"processing feature: {feature}")
+            feature_expression_series = sc.get.obs_df(self.adata, keys=feature)
+            if feature_expression_series.hasnans:
+                msg = (
+                    f"Expected numeric expression score - "
+                    f"expression data for \'{feature}\' has NaN values"
```

Comment:
Like elsewhere, let's put the variable part of the message last.  This more robustly handles long values, and generally improves greppability.  Also per convention, let's not demarcate specific values with single quotes in user-facing text.

```suggestion
                    f'expression data has NaN values for feature "{feature}"'
```

Reply:
Done `492d032`